pelican/tests/test_pelican.py

```diff
@@ -35,6 +35,22 @@ def recursiveDiff(dcmp):
     return diff
 
 
+def assertDirsEqual(left_path, right_path):
```

Comment:
I moved it out from the class as it had no reference to the test class or its instances.  

This function could be useful elsewhere too -- should this maybe be moved to ``pelican.test.support`` to be available for any other tests that need to check file diffs?

Reply:
I think, it was part of the class to follow `unittest` convention and consistency, i.e. `self.assertEqual(a, b)` etc. I don't mind taking it outside in the spirit of reusability, but in that case, yes, it should go into `pelican.tests.support`.

Reply:
Ah, I see now.  

In that case maybe it would be be better to just abstract and move the git-diff process into ``support.py``, but leave the unittest-esque method in place:

eg.
``` python
# test_pelican.py

from .support import diff_dirs_runner

class TestPelican(LoggedTestCase):

    def setUp(self):
        # ...

    def tearDown(self):
        # ...

    def assertDirsEqual(self, left_path, right_path, msg=None):
        """
        Check if the files are the same (ignoring whitespace) below both paths.
        """
        proc = diff_dirs_runner(left_path, right_path)

        out, err= proc.communicate()
        if proc.returncode != 0:
            msg = self._formatMessage(
                msg,
                "%s and %s differ:\n%s" % (left_path, right_path, err)
            )
            raise self.failureException(msg)
```

That would be more consistent with the ``unittest.TestCase class``.  Do you think something like this would be better?

Reply:
We don't need to complicate. When moving it to `support`, function doesn't need to `assert`. It can return something appropriate and caller would assert on that. I think something like this would be enough:

```python
def folders_equal(left_path, right_path):
    '''compares two folders, return if they are equal and their diff'''
    # ...
    out = proc.communicate()
    return proc.returncode == 0, out[0]
```
and in the tests:

```python
def test_something(self):
    # ...
    result, diff = folders_equal(left_path, right_path)
    self.assertTrue(result, diff)
```